debian/mariadb-server-10.4.links

```diff
@@ -7,3 +7,5 @@ usr/bin/mysqld_safe usr/bin/mariadbd-safe
 usr/bin/mysqld_safe_helper usr/bin/mariadbd-safe-helper
 usr/bin/mysqlhotcopy usr/bin/mariadb-hotcopy
 usr/bin/mysqlshow usr/bin/mariadb-show
+lib/systemd/system/mariadb.service lib/systemd/system/mysql.service
+lib/systemd/system/mariadb.service lib/systemd/system/mysqld.service
```

Comment:
Same question asked [many](https://jira.mariadb.org/browse/MDEV-15526?focusedCommentId=121830&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-121830) [many](https://github.com/MariaDB/server/pull/1172#issuecomment-481913443) times. For what purpose are symlinks needed?

There's vague comments about "align" but no substantial qualification. So substantially quantify why this should exist. What exactly is the problem being solved? What alternate solutions have you considered and what is the strengths and weaknesses of these? Show it working in a test case, preferably buildbot.

Reply:
IMHO this is the only way to support both mariadb and myslq naming (as I think that we already agreed that aliases should not be used).

Reply:
that's barely an explanation. For what purpose?

Sure, agree aliases should be dropped.

Reply:
> that's barely an explanation. For what purpose?
Don't we want this to work the exact same way:
```
$ sudo systemctl start|stop|reload|disable|enable mariadb
$ sudo systemctl start|stop|reload|disable|enable mysql
```
Do you know a better way?

Reply:
Does systemd understand this as one service with two names?
Or does this basically create two different services?

Reply:
If I understand it correctly, systemd consider it one service with two names.
See:
https://lists.freedesktop.org/archives/systemd-devel/2018-November/041658.html
https://wiki.debian.org/Teams/pkg-systemd/Packaging#systemd_unit_files_naming_and_installation

Reply:
Nope, absolutely don't want them to work the same.

Yes, the better way is remove the Alias, and add a paragraph or two on the KB about changing dependences, and maybe some of the reasons here so the rational is preserved. And most importantly, nothing else.

For the following reasons:

Continuing this equivalence with the hack as proposed (the existence in network manager packaging doesn't make it official, despite being on a systemd list) is just incurring a software debt (as past debian packaging decisions emphasise) which will fall due in the future the behaviour changes (its not documented https://www.freedesktop.org/wiki/Software/systemd/InterfaceStabilityPromise/ so not stable).

There may be cases for installing both mysql and mariadb at the same time (which of require other minor de-confliction) and there's no need for mariadb to claim the systemd service name of mysql.

As mariadb and mysql implementation paths start to diverge this equivalence status no longer applicable.

If MariaDB wants to be proud of its name, it can ask that users use "mariadb" rather using "mysql" in shadow/fake/imitation/pseudo-default-replacement.

Services can be edited to depend on the service they actually use with a small addition in /etc/systemd/systemd/{service}.d/depend_maria.conf if they assumed mysql, however because the implementation paths are diverging, lets leave it as a concious admin choice that has validated the suitability of mariadb for that service.

All these reasons sidelined so people can be ambiguous about what service they are using? Not a good justified use case IMHO.

By removing the aliases, we're paying off a bit of software debt from the people who have it already (in a broken state because of aliases), give them 2 minutes to work out the new name and move on without inuring compatibility at any cost of which Debian is so famous for.

There's certainly isn't a case for polluting RPM based packaging in addition to Debian.

Reply:
I'm confused here: This PR actually cleans up (the upstream) mariadb.service by removing the `Alias=` and moves the compat symlinks into a `debian/*` only file. So how does this pollute RPM based packaging?
If anything, it cleans up RPM based packaging.

I'm only an outsider here, so I'm not sure where the Debian packaging is supposed to happen: In the upstream Git repo or at https://salsa.debian.org/mariadb-team/mariadb-10.3 ?

Reply:
> I'm confused here: This PR actually cleans up (the upstream) mariadb.service by removing the `Alias=` and moves the compat symlinks into a `debian/*` only file. So how does this pollute RPM based packaging?
> If anything, it cleans up RPM based packaging.

I think @grooverdan made a reference to @eworm-de proposition (https://github.com/MariaDB/server/pull/1172#issuecomment-468183691)

> I'm only an outsider here, so I'm not sure where the Debian packaging is supposed to happen: In the upstream Git repo or at https://salsa.debian.org/mariadb-team/mariadb-10.3 ?

As the salsa GIT repo tries to follow as close as possible what is done upstream, if possible it is better to make changes directly in the upstream repo (and MariaDB has it's own Debian repositories https://downloads.mariadb.org/mariadb/repositories). But for some cases (problem including OpenSSL in Debian because of licenses for instance), changes has to be done directly on Debian GIT repo (salsa.debian.org).

Reply:
> Installing a init script of a different name as a service is [not recommended](https://wiki.debian.org/Teams/pkg-systemd/Packaging#systemd_unit_files_naming_and_installation). Creating symlink aliases actually makes this worse.

No, that's not true. Shipping a (static) symlink in the package is the second best thing you can do to align the sysv init script name and the service name. The best you can do (obviously) is to use the same name for the sysv init script and the service file.
In the order of preference:
1) name sysv init script and service file the same
2) ship a static symlink in the package
3) create a symlink dynamically via `Alias=`
4) do nothing

Reply:
> Nope, absolutely don't want them to work the same.

In that case (but see below), I 100% agree with you @grooverdan, not creating symlinks and removing Aliases is definitively what should be done.

But I don't think that there is a strong consensus to agree on making the `systemctl mysql` commands no longer work. I will have to discuss this with the MariaDB Foundation...

Reply:
@fauust If you remove those symlinks completely then you should rename /etc/init.d/mysql to /etc/init.d/mariadb

Reply:
Good point @mbiebl and this is why the move of not supporting `mysql` commands should be prepared better IMO. Thanks!

Reply:
> may be some other impact for users that should be taken in consideration before making this move IMHO.

If you could quantify an impact maybe it would make sense and a better answer could be given, however dawdling past GA milestones when more significant changes can occur for a still unqualified impact is just lazy.

Reply:
@grooverdan, I don't understand your point. The unique purpose of this PR is to move from 3. to 2. that @mbiebl described above (https://github.com/MariaDB/server/pull/1172#discussion_r305863850).

I am not against removing the support of the two naming, but I can't make this decision alone as I guess that this should be prepared with a communication from the MariaDB Foundation to it's users.

So I see two solutions:
- we decide to live with Aliases until we remove the two naming but we should find a solution for  MDEV-15526 on 10.3 (at least for Debian, bug is still unresolved);
- we try to remove Aliases and then we need to ship those symlinks statically with the package.

Should you have another suggestion to move forward (include https://github.com/MariaDB/server/commit/74c9872a9aae825ca6ab73feff048ff2ccb4beb8 in 10.3 branch?), please let me know.

Reply:
I'm keen to get back to a single name in a GA release without it hacks with a release notice communication (nothing much else is required). Since no effort to make a decision was made to achieve this for 10.4, 10.5 is a good goal. Maybe you could tackle that first.

I don't know why you think you can't make this decision. As such I'll leave the interim way forward decision to you (it seems practice is needed).

Reply:
So this is what I am going to do :
- propose a removal of mysql naming support in systemctl command upstream (10.5) as suggested by @grooverdan;
- fix MDEV-15526 for Debian Buster that include 10.3 branch (working on it on salsa.debian.org);
- try to have this MR also merged upstream since MDEV-15526 is not solved for 10.3 (but I believe this is not so critical that on Debian Buster since new upstream stable release is 10.4).